File: backend/src/main/java/com/sweetpos/entity/Product.java
package com.sweetpos.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "products")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product extends BaseEntity {

    @Column(nullable = false, length = 200)
    private String name;

    @Column(length = 500)
    private String description;

    @Column(unique = true, length = 50)
    private String barcode;

    @Column(length = 50)
    private String sku;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal costPrice;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal sellingPrice;

    @Column(nullable = false, precision = 5, scale = 2)
    @Builder.Default
    private BigDecimal taxRate = BigDecimal.ZERO;

    @Column(nullable = false)
    @Builder.Default
    private int stockQuantity = 0;

    @Column(nullable = false)
    @Builder.Default
    private int minStockLevel = 5;

    @Column(length = 20)
    private String unit;

    @Column(length = 500)
    private String imageUrl;

    @Column(nullable = false)
    @Builder.Default
    private boolean active = true;

    @Column(nullable = false)
    @Builder.Default
    private boolean trackInventory = true;
}

File: flutter_app/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String? description;
  final String? barcode;
  final String? sku;
  final String? categoryId;
  final String? categoryName;
  final double costPrice;
  final double sellingPrice;
  final double taxRate;
  final int stockQuantity;
  final int minStockLevel;
  final String? unit;
  final String? imageUrl;
  final bool active;
  final bool trackInventory;

  Product({
    required this.id,
    required this.name,
    this.description,
    this.barcode,
    this.sku,
    this.categoryId,
    this.categoryName,
    required this.costPrice,
    required this.sellingPrice,
    this.taxRate = 0,
    this.stockQuantity = 0,
    this.minStockLevel = 5,
    this.unit,
    this.imageUrl,
    this.active = true,
    this.trackInventory = true,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: json['id'],
        name: json['name'],
        description: json['description'],
        barcode: json['barcode'],
        sku: json['sku'],
        categoryId: json['categoryId'],
        categoryName: json['categoryName'],
        costPrice: (json['costPrice'] as num).toDouble(),
        sellingPrice: (json['sellingPrice'] as num).toDouble(),
        taxRate: (json['taxRate'] as num?)?.toDouble() ?? 0,
        stockQuantity: json['stockQuantity'] ?? 0,
        minStockLevel: json['minStockLevel'] ?? 5,
        unit: json['unit'],
        imageUrl: json['imageUrl'],
        active: json['active'] ?? true,
        trackInventory: json['trackInventory'] ?? true,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'description': description,
        'barcode': barcode,
        'sku': sku,
        'categoryId': categoryId,
        'categoryName': categoryName,
        'costPrice': costPrice,
        'sellingPrice': sellingPrice,
        'taxRate': taxRate,
        'stockQuantity': stockQuantity,
        'minStockLevel': minStockLevel,
        'unit': unit,
        'imageUrl': imageUrl,
        'active': active,
        'trackInventory': trackInventory,
      };

  bool get isLowStock => stockQuantity <= minStockLevel;
}
